src/navigation/actions.js

```diff
@@ -30,6 +34,26 @@ import { ROUTES } from './constants';
  *
  */
 
+export const goBack = () => dispatch => {
+  if (!RootNavigator.canGoBack()) BackHandler.exitApp();
+  else {
+    UIDatabase.write(() => {
+      const prescriptions = UIDatabase.objects('Prescription').filtered('status != "finalised"');
+      UIDatabase.delete('Transaction', prescriptions);
+
+      batch(() => {
+        dispatch({
+          ...NavigationActions.back(),
+          payload: {
+            prevRouteName: RootNavigator.prevRouteName(),
+          },
+        });
+        dispatch(PrescriptionActions.deletePrescription());
+      });
+    });
+  }
```

Comment:
With rendering logic dependent on navigation, I think it's a lot simpler to have this in redux - it's a few more changes to make a specific action for all the current reducers which are checking for navigating back.